File: swift/swiftfind/Sources/swiftfind/Color.swift
//
//  Color.swift
//  swiftfind
//
//  Created by Cary Clark on 5/10/25.
//  Copyright (c) 2025 Cary Clark. All rights reserved.
//

import Foundation

public enum Color {
    public static let RESET = "\u{001B}[0m"
    public static let BLACK = "\u{001B}[0;30m"
    public static let RED = "\u{001B}[0;31m"
    public static let GREEN = "\u{001B}[0;32m"
    public static let YELLOW = "\u{001B}[0;33m"
    public static let BLUE = "\u{001B}[0;34m"
    public static let MAGENTA = "\u{001B}[0;35m"
    public static let CYAN = "\u{001B}[0;36m"
    public static let WHITE = "\u{001B}[0;37m"

    public static let BOLD_BLACK = "\u{001B}[1;30m"
    public static let BOLD_RED = "\u{001B}[1;31m"
    public static let BOLD_GREEN = "\u{001B}[1;32m"
    public static let BOLD_YELLOW = "\u{001B}[1;33m"
    public static let BOLD_BLUE = "\u{001B}[1;34m"
    public static let BOLD_MAGENTA = "\u{001B}[1;35m"
    public static let BOLD_CYAN = "\u{001B}[1;36m"
    public static let BOLD_WHITE = "\u{001B}[1;37m"
}

File: csharp/CsFind/CsFindLib/Color.cs
using System;

namespace CsFindLib;

public enum Color
{
    Black,
    Red,
    Green,
    Yellow,
    Blue,
    Magenta,
    Cyan,
    White
}


public static class ColorUtil
{
    public static Color GetColorFromName(string colorName)
    {
        return Enum.TryParse<Color>(colorName, out var color) ? color : Color.Black;
    }

    public static string GetNameFromColor(Color color)
    {
        return color switch
        {
            Color.Black => "black",
            Color.Red => "red",
            Color.Green => "green",
            Color.Yellow => "yellow",
            Color.Blue => "blue",
            Color.Magenta => "magenta",
            Color.Cyan => "cyan",
            Color.White => "white",
            _ => "black"
        };
    }

    public static string ColorToConsoleColor(Color color, bool bold = false)
    {
        return color switch
        {
            Color.Black => bold ? ConsoleColor.BoldBlack : ConsoleColor.Black,
            Color.Red => bold ? ConsoleColor.BoldRed : ConsoleColor.Red,
            Color.Green => bold ? ConsoleColor.BoldGreen : ConsoleColor.Green,
            Color.Yellow => bold ? ConsoleColor.BoldYellow : ConsoleColor.Yellow,
            Color.Blue => bold ? ConsoleColor.BoldBlue : ConsoleColor.Blue,
            Color.Magenta => bold ? ConsoleColor.BoldMagenta : ConsoleColor.Magenta,
            Color.Cyan => bold ? ConsoleColor.BoldCyan : ConsoleColor.Cyan,
            Color.White => bold ? ConsoleColor.BoldWhite : ConsoleColor.White,
            _ => bold ? ConsoleColor.BoldBlack : ConsoleColor.Black
        };
    }

}
